lib/src/util.dart

```diff
@@ -41,17 +41,22 @@ Future<dynamic> retry(Future Function() f, Duration interval,
 /// Scrapes and returns the observatory URI from a string, or null if not found.
 ///
 /// Potentially useful as a means to extract it from log statements.
-Uri? extractObservatoryUri(String str) {
-  const kObservatoryListening = 'Observatory listening on ';
-  final msgPos = str.indexOf(kObservatoryListening);
-  if (msgPos == -1) return null;
-  final startPos = msgPos + kObservatoryListening.length;
-  final endPos = str.indexOf(RegExp(r'(\s|$)'), startPos);
-  try {
-    return Uri.parse(str.substring(startPos, endPos));
-  } on FormatException {
-    return null;
+@Deprecated('Use extractVMServiceUri instead')
```

Comment:
I don't think this is in the public API so we should be able to skip a deprecation period.

Are you aware of a `src` import with usage of this from outside the package?

Reply:
For some reason I got the impression this was used elsewhere, but it definitely isn't. I'll remove this.